lib/payment_processor/clients/braintree/error_processing.rb

```diff
@@ -0,0 +1,78 @@
+module PaymentProcessor
+  module Clients
+    module Braintree
+      class ErrorProcessing
+
+        # Collection of error codes, generated by braintree for invalid
+        # customer/credit card transaction data.
+        #
+        # See https://developers.braintreepayments.com/reference/general/validation-errors/all/ruby
+        # for code definitions.
+        #
+        USER_ERROR_CODES = [
+          (81801..81813),
+          (81501..81503),
+          (91814..91826),
+          (81604..81608),
+          (81613..81616),
+          (81709..81717),
+          (91723..91726),
+          81827,
+          91803,
+          81601,
+          81723,
+          81703,
+          81706,
+          81707,
+          81736,
+          81737,
+          81528,
+          81509].map{|code| code.is_a?(Range) ? code.to_a : code }
+                .flatten
+                .freeze
+
+        FILTER = -> (error) { USER_ERROR_CODES.include?( error.code.to_i ) }
+
+        def initialize(response)
+          @response = response
+        end
+
+        def process
+          if has_user_errors
+            return user_errors
+          else
+            raise_system_errors
+          end
+        end
+
+        private
+
+        def has_user_errors
+          errors.any?(&FILTER)
```

Comment:
you're running the filter twice, if you wanted to run it once you could just do `user_errors.size > 0` but I think it really doesn't matter